examples/README.md

```diff
@@ -17,7 +17,7 @@ A comprehensive travel assistant that demonstrates:
 The travel agent automatically discovers and uses all memory tools available from the client:
 
 1. **search_memory** - Search through previous conversations and stored information
-2. **get_working_memory** - Check current session state, stored memories, and data
+2. **get_or_create_working_memory** - Check current session state, stored memories, and data
 3. **add_memory_to_working_memory** - Store important information as structured memories
```

Comment:
It's confusing that we have three ways of creating long-term memories: background processing from messages, adding them lazily to a working memory session, and eagerly creating them. Rename the different tools to more clearly explain what they do, and deprecate the old ones:
- add_memory_to_working_memory -> lazily_create_long_term_memory - Lazily create a long-term memory by adding it to working memory (does not require an immediate network request; does require saving working memory afterward)
- create_long_term_memory (should be on this list?) -> eagerly_create_long_term_memory - Eagerly create a long-term memory by making an API request